source/standalone/workflows/ray/isaac_ray_util.py

```diff
@@ -0,0 +1,409 @@
+            directory. Can be used to avoid losing logs in the /tmp directory. Defaults to None.
+
+    Raises:
+        ValueError: If the job is unable to start, or throws an error. Most likely to happen
+            due to running out of memory.
+
+    Returns:
+        Relevant information from the job
+    """
+    start_time = datetime.now().strftime("%H:%M:%S.%f")
+    result_details = [f"{identifier_string}: ---------------------------------\n"]
+    result_details.append(f"{identifier_string}:[INFO]: Invocation {job_cmd} \n")
+    node_id = ray.get_runtime_context().get_node_id()
+    result_details.append(f"{identifier_string}:[INFO]: Ray Node ID: {node_id} \n")
+
+    if test_mode:
+        import torch
+
+        try:
+            result = subprocess.run(
+                ["nvidia-smi", "--query-gpu=name,memory.free,serial", "--format=csv,noheader,nounits"],
+                capture_output=True,
+                check=True,
+                text=True,
+            )
+            output = result.stdout.strip().split("\n")
+            for gpu_info in output:
+                name, memory_free, serial = gpu_info.split(", ")
+                result_details.append(
+                    f"{identifier_string}[INFO]: Name: {name}|Memory Available: {memory_free} MB|Serial Number"
+                    f" {serial} \n"
+                )
+
+            # Get GPU count from PyTorch
+            num_gpus_detected = torch.cuda.device_count()
+            result_details.append(f"{identifier_string}[INFO]: Detected GPUs from PyTorch: {num_gpus_detected} \n")
+
+            # Check CUDA_VISIBLE_DEVICES and count the number of visible GPUs
+            cuda_visible_devices = os.environ.get("CUDA_VISIBLE_DEVICES")
+            if cuda_visible_devices:
+                visible_devices_count = len(cuda_visible_devices.split(","))
+                result_details.append(
+                    f"{identifier_string}[INFO]: GPUs visible via CUDA_VISIBLE_DEVICES: {visible_devices_count} \n"
+                )
+            else:
+                visible_devices_count = len(output)  # All GPUs visible if CUDA_VISIBLE_DEVICES is not set
+                result_details.append(
+                    f"{identifier_string}[INFO]: CUDA_VISIBLE_DEVICES not set; all GPUs visible"
+                    f" ({visible_devices_count}) \n"
+                )
+
+            # If PyTorch GPU count disagrees with nvidia-smi, reset CUDA_VISIBLE_DEVICES and rerun detection
+            if num_gpus_detected != len(output):
+                result_details.append(
+                    f"{identifier_string}[WARNING]: PyTorch and nvidia-smi disagree on GPU count! Re-running with all"
+                    " GPUs visible. \n"
+                )
+                result_details.append(f"{identifier_string}[INFO]: This shows that GPU resources were isolated.\n")
+                os.environ["CUDA_VISIBLE_DEVICES"] = ",".join([str(i) for i in range(len(output))])
```

Comment:
This seems a little sketchy to modify the user's envvar, but I'm not super familiar

Reply:
Only executed as part of test - it's ephemeral to this single run and would not persist. It's a useful sanity check